src/v/cluster/types.cc

```diff
@@ -320,6 +321,7 @@ adl<cluster::topic_configuration>::from(iobuf_parser& in) {
     cfg.properties.timestamp_type
       = adl<std::optional<model::timestamp_type>>{}.from(in);
     cfg.properties.segment_size = adl<std::optional<size_t>>{}.from(in);
+    cfg.properties.source_topic = adl<std::optional<model::topic>>{}.from(in);
```

Comment:
Looks like it might, what do you recommend?

Reply:
It looks like the thing to do is add a test that shows that after an upgrade existing topic_configurations that are serialized to disk can continue to be read. It looks like it is a little more complicated than normal because topic_configuration seems to also be embedded inside topic_configuration_assignment, and there may also be some other places like that.

Reply:
I moved this parameter to be part of `topic_metadata` side stepping this issue